src/vsr/replica.zig

```diff
@@ -5879,15 +5950,9 @@ pub fn ReplicaType(
                     break :repair_min 0;
                 }
 
-                const op_checkpoint_trigger =
```

Comment:
Hm, why are we looking at `checkpoint_quorum` here?  Shouldn't we just look at our `op`?

If op is beyond `op_prepare_max`, we can drop the old checkpoint. Otherwise, can keep it. 

That way, only the primary would call `checkpoint_quorum`, and we will just (implicitly, by observring a prepare beyond prepare_max) re-use that information.

Reply:
https://github.com/tigerbeetle/tigerbeetle/pull/2104/commits/3d24e4a6d05ec1ba516fa568ef9e4f5ebf721fd4 simplifies checkpoint_quorum() such that it is only invoked from the primary!